src/input/isf.c

```diff
@@ -0,0 +1,529 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <config.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <ctype.h>
+#include <string.h>
+#include <stdint.h>
+#include <math.h>
+#include <libsigrok/libsigrok.h>
+#include "libsigrok-internal.h"
+
+#define LOG_PREFIX "input/isf"
+
+#define CHUNK_SIZE	(4 * 1024 * 1024)
+
+/* Number of bytes required to process header */
+#define MIN_INITIAL_OFFSET 512
+
+#define HEADER_ITEMS_PARAMETERS 10
+#define MAX_CHANNEL_NAME_SIZE 32
+
+enum byteorder{
+    LSB = 0,
+    MSB = 1,
+};
+
+enum format{
+    RI = 0,
+    FP = 1,
+};
+
+enum waveform_type{
+    ANALOG = 0,
+    RF_FD = 1,
+};
+
+union floating_point{
+    float f;
+    uint32_t i;
+};
+
+struct context{
+    gboolean started;
+    gboolean found_header;
+    gboolean create_channel;
+    float yoff;
+    float yzero;
+    float ymult;
+    float xincr;
+    int bytnr;
+    int byte_order;
+    int bn_fmt;
+    int wfmtype;
+    char channel_name[MAX_CHANNEL_NAME_SIZE];
+};
+
+enum header_items_enum{
+    YOFF = 0,
+    YZERO = 1,
+    YMULT = 2,
+    XINCR = 3,
+    BYTNR = 4,
+    BYTE_ORDER = 5,
+    BN_FMT = 6,
+    WFID = 7,
+    WFMTYPE = 8,
+    ENCODING = 9,
+};
+
+static const char *header_items[] = {
+    [YOFF] = "YOFF ",
+    [YZERO] = "YZERO ",
+    [YMULT] = "YMULT ",
+    [XINCR] = "XINCR ",
+    [BYTNR] = "BYT_NR ",
+    [BYTE_ORDER] = "BYT_OR ",
+    [BN_FMT] = "BN_FMT ",
+    [WFID] = "WFID ",
+    [WFMTYPE] = "WFMTYPE ",
+    [ENCODING] = "ENCDG ",
+};
+
+
+static char *find_item(GString *buf, const char *item)
+{
+    return g_strstr_len(buf->str, buf->len, item);
+}
+
+static gboolean has_header(GString *buf)
+{
+    return buf->len >= MIN_INITIAL_OFFSET;
+}
+
+static char *find_data_section(GString *buf)
+{
+    const char *curve = "CURVE #";
+
+    char *data_ptr;
+    size_t offset, data_length;
+
+    data_ptr = g_strstr_len(buf->str, buf->len, curve);
+    if(data_ptr == NULL)
+        return NULL;
+
+    data_ptr += strlen(curve);
+    offset = data_ptr - buf->str;
+    if(offset >= buf->len)
+        return NULL;
+
+    data_length = (size_t) *data_ptr - 48;
+    data_ptr += 1 + data_length;
+    offset = data_ptr - buf->str;
+
+    if(offset >= buf->len)
+        return NULL;
+
+    return data_ptr;
+}
+
+static void extract_channel_name(struct context *inc, const char *beg)
+{
+    size_t i, channel_ix;
+
+    channel_ix = 0;
+    /* Isf wfid looks something like WFID "Ch1, ..."; thus we must skip character '"' */
```

Comment:
Sorry for being inactive for a long time. This comment was supposed to mean that WFID follows this pattern in ISF format. So Isf was not a typing error, in case I understand your review correctly.

Reply:
suggest writing acronyms  in all-caps ?, "ISF WFID" looks a lot less accidental